src/store/CategoricalDisplayStore.js

```diff
@@ -177,8 +177,16 @@ export default class CategoricalDisplayStore {
             filteredItems.unshift(activeCategory);
         }
 
+        /**
+        * when search bar is empty or it's the first rendering
+        * lastFilteredItems should contain all data
+        */
+        this.lastFilteredItems = !this.lastFilteredItems.length || this.filterText === ''
```

Comment:
can you spend more time to make this simpler. It should simply communicate that it will return the last available search result if there is no available result.